File: csv_detective/formats/mois_de_lannee.py
from unidecode import unidecode

proportion = 1
tags = ["fr", "temp"]
labels = ["mois", "month"]

mois = {
    "janvier",
    "fevrier",
    "mars",
    "avril",
    "mai",
    "juin",
    "juillet",
    "aout",
    "septembre",
    "octobre",
    "novembre",
    "decembre",
    "jan",
    "fev",
    "mar",
    "avr",
    "mai",
    "jun",
    "jui",
    "juil",
    "aou",
    "sep",
    "sept",
    "oct",
    "nov",
    "dec",
}


def _is(val):
    """Renvoie True si les champs peuvent être des mois de l'année"""
    if not isinstance(val, str):
        return False
    val = unidecode(val.lower())
    return val in mois


_test_values = {
    True: ["JUIN", "décembre"],
    False: ["november"],
}

File: rust/src/formats/mois_de_lannee.rs
use crate::value::Value;

const VALID: &[&str] = &[
    "janvier", "fevrier", "mars", "avril", "mai", "juin",
    "juillet", "aout", "septembre", "octobre", "novembre", "decembre",
    "jan", "fev", "mar", "avr", "jun", "jui", "juil", "aou",
    "sep", "sept", "oct", "nov", "dec",
];

pub fn detect(val: &str) -> Option<()> {
    // normalize unicode + lowercase
    let lower = val.to_lowercase();
    let normalized = normalize_accents(&lower);
    if VALID.contains(&normalized.as_str()) {
        Some(())
    } else {
        None
    }
}

fn normalize_accents(s: &str) -> String {
    s.replace('é', "e")
        .replace('è', "e")
        .replace('ê', "e")
        .replace('û', "u")
        .replace('ù', "u")
        .replace('â', "a")
        .replace('à', "a")
        .replace('î', "i")
        .replace('ô', "o")
}

pub fn test(val: &Value) -> bool {
    detect(val.raw()).is_some()
}
